### PR TITLE
DOC: Add whatsnew entry for v0.23

### PR DESCRIPTION
**Pull request by greglucas**
This is in preparation for the v0.23 release. Let me know if I missed any features and if we should add anything else to this list.

### CODE REVIEW

**Comment by rcomer** [comment]
#2280 expired some deprecations.  Should that be noted in the deprecations section?

Maybe a note that we now officially support python 3.12?

Something about the whereabouts of the feature download script might be useful for those who need to use it (#2263).

Over in Iris, we have started including a reviewer credit for the more substantial changes (e.g. [v3.8 features](https://scitools-iris.readthedocs.io/en/stable/whatsnew/3.8.html#features)).  What do you think about doing that here?  I definitely think you deserve some credit for the `GridLiner` and `FeatureArtist` changes!

**Comment by greglucas** [comment]
> https://github.com/SciTools/cartopy/pull/2280 expired some deprecations. Should that be noted in the deprecations section?

> Maybe a note that we now officially support python 3.12?

> Something about the whereabouts of the feature download script might be useful for those who need to use it (https://github.com/SciTools/cartopy/pull/2263).

Thanks! I added all of these.

> Over in Iris, we have started including a reviewer credit for the more substantial changes (e.g. [v3.8 features](https://scitools-iris.readthedocs.io/en/stable/whatsnew/3.8.html#features)). What do you think about doing that here? I definitely think you deserve some credit for the GridLiner and FeatureArtist changes!

That is nice, I think that'd be a great idea. It'd also be good if we start building this file as we go rather than having someone go through old commits and PRs to see what was changed :) I just asked questions during the review, you found the answers! Much less work for me, so I appreciate all the effort you put into updating those features.

File: docs/source/whatsnew/v0.23.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  add and remove gridlines and fewer internal draws for better performance. (:pull:`2249`, :pull:`2252`)
+
+* The cartopy feature download script (useful for downloading Natural Earth Features
+  for offline use) can be invoked within the package directly
+  using :code:`python -m cartopy.feature.download`, or through the installed command
+  line interface using :code:`cartopy-feature-download`. (:pull:`2263`)
+
+* The Stamen Maps API is no longer available. There is a new class StadiaMaps
+  that can be used to access the Stadia Maps API which contains the
+  Stamen styled tiles. (:pull:`2269`)
```
Comment (rcomer) [inline docs/source/whatsnew/v0.23.rst:33]:
```suggestion
* Ruth Comer turned the ``GridLiner`` into a Matplotlib ``Artist`` making it easier to
  add and remove gridlines and fewer internal draws for better performance. (:pull:`2249`, :pull:`2252`)

* The cartopy feature download script (useful for downloading Natural Earth Features
  for offline use) can be invoked within the package directly
  using :code:`python -m cartopy.feature.download`, or through the installed command
  line interface using :code:`cartopy-feature-download`. (:pull:`2263`)

* The Stamen Maps API is no longer available. There is a new class ``StadiaMaps``
  that can be used to access the Stadia Maps API which contains the
  Stamen styled tiles. (:pull:`2269`)
```
Just for consistency with other points.

File: docs/source/whatsnew/v0.23.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
+Version 0.23 (March XX, 2024)
+=============================
+
+Cartopy has been relicensed from LBPL-3 to BSD-3-Clause. All contributions
```
Comment (rcomer) [inline docs/source/whatsnew/v0.23.rst:4]:
```suggestion
Cartopy has been relicensed from LGPL-3 to BSD-3-Clause. All contributions
```

**Review by rcomer — COMMENTED** [review]
> It'd also be good if we start building this file as we go rather than having someone go through old commits and PRs to see what was changed :)

Yeah it's usually easier to describe something when you only just did it too!

**Review by rcomer — APPROVED** [review]

**Comment by rcomer** [comment]
I think this is good to go if no-one else has comments.  @greglucas do you want to squash the commits?  Feel free to self-merge either way.

**Comment by greglucas** [comment]
I have set the release date for tomorrow April 4th in this change note now. That might be optimistic, but I think this PR and the numpy 2.0 build https://github.com/SciTools/cartopy/pull/2366 are the only two PRs needed before we can release, so if those go in I think we should release to prepare for Python 3.12+ and numpy 2.0+.

**Comment by greglucas** [comment]
🤔 There is a lot of flakiness going on in the tests right now too. I had to rerun failed jobs a few times before they succeeded, and they are on image comparison tests without network requirements which seems odd to me.

File: docs/source/whatsnew/v0.23.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
+Version 0.23 (April 4, 2024)
+============================
+
+Cartopy has been relicensed from LGPL-3 to BSD-3-Clause. All contributions
+to Cartopy are now under the BSD-3-Clause license.
+
+Python 3.12 is now supported and the new minimum supported versions
```
Comment (rcomer) [inline docs/source/whatsnew/v0.23.rst:7]: Should we mention numpy 2.0 in here too?

Comment (greglucas) [reply docs/source/whatsnew/v0.23.rst]: Yes, good suggestion. Just added that note.

**Pull request merged**